app/controllers/social_networking/profile_pages_controller.rb

```diff
@@ -43,18 +27,14 @@ def set_member_profiles
     end
 
     def set_current_profile
-      id = params[:id] || current_participant.id
-      @profile = Profile.find_or_create_by(
-        participant_id: id, active: true) do |profile|
-        begin
-          SharedItem.create(
-            item: profile,
-            action_type: Profile::Actions.created)
-        rescue ActiveRecord::StatementInvalid
-          logger.info("Shared item already created for existing profile.")
-        end
+      if params[:id]
+        @profile = Profile.find(params[:id])
+      else
+        @profile = Profile
+                   .find_or_initialize_by(
+                     participant_id: current_participant.id)
+        @profile.update_attributes(active: true) unless @profile.active
```

Comment:
I was just thinking that is a lot of unnecessary updates.  Do you think that concern isn't necessary?